src/main/java/com/es/phoneshop/web/servlets/ProductDetailsPageServlet.java

```diff
@@ -0,0 +1,30 @@
+package com.es.phoneshop.web.servlets;
+
+import com.es.phoneshop.service.ArrayListProductService;
+import com.es.phoneshop.service.ProductService;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ProductDetailsPageServlet extends HttpServlet {
+
+    ProductService productService = new ArrayListProductService();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String productCode = request.getParameter("productCode");
+        System.out.println(productCode);
+        String productDetails = productService.getProductDetailsByCode(productCode);
+        if (productDetails == null) {
```

Comment:
You should never return `null`, but throw `ProductNotFoundException`. Map this exception to 404 HTTP code in `web.xml` and bind it to custom `404.jsp`